Classes/Notifications/NotificationCell2.swift

```diff
@@ -0,0 +1,199 @@
+//
+//  NotificationCell2.swift
+//  Freetime
+//
+//  Created by Ryan Nystrom on 6/8/18.
+//  Copyright © 2018 Ryan Nystrom. All rights reserved.
+//
+
+import UIKit
+import SnapKit
+import StyledTextKit
+
+protocol NotificationCell2Delegate: class {
+    func didTapRead(cell: NotificationCell2)
+    func didTapWatch(cell: NotificationCell2)
+    func didTapMore(cell: NotificationCell2, sender: UIView)
+}
+
+final class NotificationCell2: SelectableCell {
+
+    public static let inset = UIEdgeInsets(
+        top: NotificationCell2.topInset + NotificationCell2.headerHeight + Styles.Sizes.rowSpacing,
+        left: Styles.Sizes.icon.width + 2*Styles.Sizes.columnSpacing,
+        bottom: NotificationCell2.actionsHeight,
+        right: Styles.Sizes.gutter
+    )
+    public static let topInset = Styles.Sizes.rowSpacing
+    public static let headerHeight = ceil(Styles.Text.secondary.preferredFont.lineHeight)
+    public static let actionsHeight = Styles.Sizes.gutter + 4*Styles.Sizes.rowSpacing
+
+    private weak var delegate: NotificationCell2Delegate?
+    private let iconImageView = UIImageView()
+    private let dateLabel = ShowMoreDetailsLabel()
+    private let detailsLabel = UILabel()
+    private let textView = StyledTextView()
+    private let stackView = UIStackView()
+    private let commentButton = UIButton()
+    private let readButton = UIButton()
+    private let watchButton = UIButton()
+    private let moreButton = UIButton()
+
+    override init(frame: CGRect) {
```

Comment:
I think the `init` is too large. I would split it up in smaller `functions` and call these `functions` in the `init`. 
e.g:

```swift

override init(frame: CGRect) {
   ...
   
   configureCommentButton()
   ...
}

private func configureCommentButton() {
    contentView.addSubview(commentButton)

    commentButton.titleLabel?.font = font
    commentButton.isUserInteractionEnabled = false
    commentButton.tintColor = grey
    commentButton.setTitleColor(grey, for: .normal)
    commentButton.contentEdgeInsets = UIEdgeInsets(top: 0, left: 0, bottom: -2, right: 0)
    commentButton.titleEdgeInsets = UIEdgeInsets(top: -4, left: 2, bottom: 0, right: 0)
    commentButton.setImage(UIImage(named: "comment-small")?.withRenderingMode(.alwaysTemplate), for: .normal)
    commentButton.contentHorizontalAlignment = .left

    commentButton.snp.makeConstraints { make in
        make.width.equalTo(actionsHeight)
    }
}
```

Reply:
Unless the functions are going to be used again, I think this gets too confusing and is error prone.

What happens if one of these configs depends on another view being added first? Now they implicitly depend on each other w/out any way of knowing the order of operations.

Putting them all inline makes it extremely simple, easy to understand, and shows that this setup isn't ever called again.